schematic/store/synapse.py

```diff
@@ -65,18 +69,31 @@ class ManifestDownload(object):
     syn: synapseclient.Synapse
     manifest_id: str
 
-    def _download_manifest_to_folder(self):
+    def _download_manifest_to_folder(self) -> File:
         """
         try downloading a manifest to local cache or a given folder
         manifest
         Return: 
-            manifest_data: A new Synapse Entity object of the appropriate type
+            manifest_data: synapse file entity
         """
-         # TO DO: potentially deprecate the if else statement because "manifest_folder" key always exist in config
-        if CONFIG["synapse"]["manifest_folder"]:
+        # TO DO: potentially deprecate the if else statement because "manifest_folder" key always exist in config
+        # on AWS, to avoid overriding manifest, we download the manifest to a temporary folder
+        if "SECRETS_MANAGER_SECRETS" in os.environ:
```

Comment:
This is a really complex series of if/elif/else statements, and really hard to parse. Is there any way to clean this up and make it easier to understand? If not I would recommend adding more documentation.

Reply:
+1. This chunk of code seems to be logic for `_determine_download_location`.  The main question I have is this: 

* Why does "SECRETS_MANAGER_SECRETS" existing in `os.environ` take precedence over `CONFIG['synapse']['manifest']`?

Reply:
@thomasyu888 That's because the config on AWS also has "manifest_folder" key. If I have `if CONFIG["synapse"]["manifest_folder"]` takes precedence, then the logic of downloading to a temporary folder can't be triggered. 

I think an alternative solution is to use a different config for deploying to AWS. Since in the future, we would use the develop branch to deploy to AWS, we can't add a different key in config.yml to indicate deployment. But maybe we could have a separate config.yml just for deployment? Will that bring in more complexity?

Reply:
Thanks - this is helpful. Similar to how AWS does credentials and profiles, maybe something similar can be done here? Can open a jira ticket to discuss.

I don't know enough of the codebase, but potentially you can see if there are duplicated keys in your CONFIG and look into simplifying functions to take in the the 'synapse' or 'aws' parameter instead of if-else statements.  You could have different "profile" configurations like you said when deploying differently.   I'll defer this decision to the FAIR team.

Reply:
@andrewelamb I discussed this issue with Milen in our 1:1. He suggested that let's leave this code here for now and see if the solution works for fixing your bug. If it does, we could circle back to this issue and discuss with the team what would be a good way to handle downloading manifests on AWS v.s. downloading manifests locally